### PR TITLE
force npm install before npm test

### PR DESCRIPTION
**Pull request by EndangeredMassa**
Force a clean `npm install` to succeed before running the tests.

---

Addresses: https://github.com/EndangeredMassa/npub/issues/24

### CODE REVIEW

File: src/publish/index.coffee [hunk 1]
```diff
@@ -47,48 +47,51 @@ module.exports = (dir, log, config, version, testCommand) ->
     verify dir, (error) ->
       endIf(error)
 
-      test dir, log, npm, testCommand, (error) ->
+      npm.install config.registry, (error) ->
```
Comment (jkrems) [inline src/publish/index.coffee]: It gets really hard to see what actually changed in here. At some point an `async.series` or similar might help make it easier to follow/track.

**Comment by jkrems** [comment]
LGTM

File: src/publish/index.coffee (hunk 1 above)
Comment (EndangeredMassa) [inline src/publish/index.coffee]: Agreed, just been putting it off.

**Comment by EndangeredMassa** [comment]
published in [2.0.0](https://github.com/EndangeredMassa/npub/releases/tag/v2.0.0)

**Pull request merged**